test/common/extensions/filters/http/platform_bridge/platform_bridge_filter_integration_test.cc

```diff
@@ -68,67 +68,67 @@ class PlatformBridgeIntegrationTest : public testing::TestWithParam<Network::Add
-  envoy_http_filter platform_filter_1{};
-  filter_invocations invocations_1{};
-  platform_filter_1.static_context = &invocations_1;
-  platform_filter_1.init_filter = [](const void* context) -> const void* {
-    envoy_http_filter* c_filter = static_cast<envoy_http_filter*>(const_cast<void*>(context));
-    filter_invocations* invocations =
-        static_cast<filter_invocations*>(const_cast<void*>(c_filter->static_context));
-    invocations->init_filter_calls++;
-    return invocations;
-  };
-  platform_filter_1.on_response_data = [](envoy_data c_data, bool,
-                                          const void* context) -> envoy_filter_data_status {
-    filter_invocations* invocations = static_cast<filter_invocations*>(const_cast<void*>(context));
-    invocations->on_response_data_calls++;
-    return {kEnvoyFilterDataStatusContinue, c_data, nullptr};
-  };
-  platform_filter_1.release_filter = [](const void*) {};
-
-  envoy_http_filter platform_filter_2{};
-  filter_invocations invocations_2{};
-  platform_filter_2.static_context = &invocations_2;
-  platform_filter_2.init_filter = [](const void* context) -> const void* {
-    envoy_http_filter* c_filter = static_cast<envoy_http_filter*>(const_cast<void*>(context));
-    filter_invocations* invocations =
-        static_cast<filter_invocations*>(const_cast<void*>(c_filter->static_context));
-    invocations->init_filter_calls++;
-    return invocations;
-  };
-  platform_filter_2.on_response_data = [](envoy_data c_data, bool end_stream,
-                                          const void* context) -> envoy_filter_data_status {
-    filter_invocations* invocations = static_cast<filter_invocations*>(const_cast<void*>(context));
-    invocations->on_response_data_calls++;
-    if (!end_stream) {
-      c_data.release(c_data.context);
-      return {kEnvoyFilterDataStatusStopIterationAndBuffer, envoy_nodata, nullptr};
-    } else {
-      return {kEnvoyFilterDataStatusResumeIteration, c_data, nullptr};
-    }
-  };
-  platform_filter_2.release_filter = [](const void*) {};
-
-  customInit(&platform_filter_1, &platform_filter_2);
-
-  auto response = codec_client_->makeHeaderOnlyRequest(default_request_headers_);
-
-  // Wait for frames to arrive upstream.
-  ASSERT_TRUE(fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_));
-  ASSERT_TRUE(fake_upstream_connection_->waitForNewStream(*dispatcher_, upstream_request_));
-  ASSERT_TRUE(upstream_request_->waitForEndStream(*dispatcher_));
-
-  upstream_request_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, false);
-
-  upstream_request_->encodeData(100, false);
-  upstream_request_->encodeData(100, true);
-
-  // Wait for frames to arrive downstream.
-  ASSERT_TRUE(response->waitForEndStream());
-
-  EXPECT_TRUE(response->complete());
-  EXPECT_EQ("200", response->headers().Status()->value().getStringView());
-}
+// TEST_P(PlatformBridgeIntegrationTest, MultipleFilters) {
```

Comment:
@goaway given how request and responses are passed in threading contexts in the integration text framework, I am having trouble getting the filters to run in the thread of their own dispatcher without stalling the test. 

If you are ok with it, could we maybe land this PR, and I can untangle the threading issue in this integration test in a subsequent PR?

Reply:
I can open a ticket to track

Reply:
Yes, please open a tracking issue/ticket.

Reply:
https://github.com/envoyproxy/envoy-mobile/issues/1533